### ISSUE TITLE
Add alerts on slack for new "Security" issues

### ISSUE THREAD

**Issue by potiuk** [apache/airflow#44382]
There is a "security" tab in the airflow repository where code scanning produces new issues discovered in our code. 

In order to drag attention to it, we should have an automation to post slack messages in a private "security" channel - this, similarly as in case of main failures - might help us with more "group" handling of noticing and handling such security reports.

**Comment by potiuk** [comment]
cc: @amoghrajesh - when you have time :)

**Comment by eladkal** [comment]
Maybe we can find a way for the scanner to run on the PR directly before we merge it?

**Comment by potiuk** [comment]
> Maybe we can find a way for the scanner to run on the PR directly before we merge it?

Yes. good point https://github.com/apache/airflow/blob/main/.github/workflows/codeql-analysis.yml likely need to be updated to work on `pull request` as well. I am not sure if it is going to work on pull requests from forks - but there does not seem to be a limitation for it mention and it needs only 'read-content" permission: https://docs.github.com/en/code-security/code-scanning/managing-code-scanning-alerts/triaging-code-scanning-alerts-in-pull-requests

**Comment by potiuk** [comment]
https://github.com/apache/airflow/pull/44404 -> seems to work out of the box.
